clean up component dependencies of lia

andres-erbsen opened the pull request:
- [x] Added **changelog**.
- [x] Added / updated **documentation**.
- [x] Opened **overlay** pull requests.
  - [ ] https://github.com/PrincetonUniversity/VST/pull/821
  - [x] https://github.com/mit-plv/bedrock2/pull/461
  - [x] https://github.com/mit-plv/fiat-crypto/pull/2089

proux01 reviewed (changes requested):
Sounds reasonable at first glance

andres-erbsen commented on doc/stdlib/index-list.html.template:
```diff
@@ -233,68 +207,58 @@ through the <tt>From Stdlib Require Import</tt> command.</p>
-    Also contains the migromega tactic that can be loaded with
-    From Stdlib Require Import Lia.
+    Integer-arithmetic properties
   </dt>
```
Updated. Most of the information is now split between entries for integers and primitive-int. I put back more mention of proofs and the suggested Require Import here. I removed the historical note because it does not seem relevant and important enough to be on the front page.

proux01 commented:
> how to handle the remaining overlay.

IIUC, this is only VST. You can probably ask them to merge it once their CI confirms it's indeed backward compatible.

andres-erbsen commented on doc/stdlib/index-list.html.template:
```diff
@@ -233,69 +209,61 @@ through the <tt>From Stdlib Require Import</tt> command.</p>
   </dd>
 
   <dt> <a name="zarith"></a><b>ZArith</b>:
-    Binary encoding of integers.
-    This binary encoding was initially developped to enable effective
-    computations, compared to the unary encoding of nat. Proofs were then added
-    making the types usable for mathematical proofs, although this was not
-    the initial intent. If even-more efficient computations are needed, look
-    at the <a href="#primitive-int">primitive-int</a> package below for 63 bits machine arithmetic
-    or the coq-bignums package for arbitrary precision machine arithmetic.
-    Everything can be imported with From Stdlib Require Import ZArith.
-    Also contains the migromega tactic that can be loaded with
-    From Stdlib Require Import Lia.
```
I responded [here](https://github.com/rocq-prover/stdlib/pull/150#discussion_r2142584214)

andres-erbsen commented:
This is now rebased and cleaned up. I intend to merge tomorrow. (The new check in the last commit was important to for checking that the rebase left subcomponent files in an instructive state.)
